aiokafka/conn.py

```diff
@@ -94,7 +94,12 @@ async def create_conn(
         sasl_kerberos_domain_name=sasl_kerberos_domain_name,
         sasl_oauth_token_provider=sasl_oauth_token_provider,
         version_hint=version_hint)
-    await conn.connect()
+    try:
+        await conn.connect()
+    except Exception:
+        # Cleanup to prevent `Unclosed AIOKafkaConnection` if we failed to connect here
+        conn.close()
```

Comment:
Shouldn't be this logic inside `AIOKafkaConnection.connect()`, so that it maintains invariant too?

Reply:
Not sure about that, because `AIOKafkaConnection.connect()` is not being used directly, only via `create_conn` factory. Though, I can move this wrapping logic if you think that it should be more appropriate to have it there.